types/stylelint/index.d.ts

```diff
@@ -146,9 +146,9 @@ declare module 'stylelint' {
 
 	export type StylelintRuleMessages = Record<string, string | ((...args: any[]) => string)>;
 
-	export type StylelintRule = ((
-		primaryOption: any,
-		secondaryOptions: Record<string, any>,
+	export type StylelintRule<P = any, S = any> = ((
+		primaryOption: P,
+		secondaryOptions: Record<string, S>,
```

Comment:
[note] **P** - **P**rimary, **S** - **S**econdary

Reply:
Why do not use full words? :smile:

Reply:
I follow the existing style, but is using full words common? (I'm open to change)

https://github.com/stylelint/stylelint/blob/dddf8568cfdb2ec65f49164c1a5b67379d8756a8/types/stylelint/index.d.ts#L12

https://github.com/stylelint/stylelint/blob/dddf8568cfdb2ec65f49164c1a5b67379d8756a8/types/stylelint/index.d.ts#L51

Reply:
Let's follow the existing style. We can revisit the lot if it becomes an issue.